src/feature/RansacFeatureSetMatcher.cpp

```diff
@@ -51,26 +60,82 @@ double RansacFeatureSetMatcher::matchSets(const std::vector<InterestPoint *> &re
-   std::vector< std::pair<InterestPoint *, InterestPoint *> > possibleCorrespondences;
-   for(unsigned int i = 0; i < data.size(); i++){
-	double minCorrespondenceDistance = 1e17;
-	unsigned int minCorrespondenceIndex = 0;
-	for(unsigned int j = 0; j < reference.size(); j++){
-	    double distance = data[i]->getDescriptor()->distance(reference[j]->getDescriptor());
-	    if(distance < minCorrespondenceDistance){
-		minCorrespondenceDistance = distance;
-		minCorrespondenceIndex = j;
-	    }
-	}
-	if(minCorrespondenceDistance < m_distanceThreshold){
-	    possibleCorrespondences.push_back(std::make_pair(data[i], reference[minCorrespondenceIndex]));
-	}
+    std::vector< std::pair<InterestPoint *, InterestPoint *> > possibleCorrespondences;
+    std::vector<IndexedDistance> minDistances;
+
+    if (!m_flann){
+        for(unsigned int i = 0; i < data.size(); i++){
+            double minCorrespondenceDistance = 1e17;
+            unsigned int minCorrespondenceIndex = 0;
+            for(unsigned int j = 0; j < reference.size(); j++){
+                double distance = data[i]->getDescriptor()->distance(reference[j]->getDescriptor());
+                if(distance < minCorrespondenceDistance){
+                    minCorrespondenceDistance = distance;
+                    minCorrespondenceIndex = j;
+                }
+             }
+
+            if(minCorrespondenceDistance < m_distanceThreshold){
+                possibleCorrespondences.push_back(std::make_pair(data[i], reference[minCorrespondenceIndex]));
+                minDistances.push_back(std::make_pair(minDistances.size(), minCorrespondenceDistance));
+            }
+        }
+    }
+    else { /* m_flann */
+        // Number of nearest neighbours to use
+        size_t nn = 1;
+   
+        // Create and fill the flann matrices from the data and reference descriptors
+        std::vector<double> data_descriptors;
+        std::vector<double> reference_descriptors;
+        std::vector<int> indices(data.size()*nn);
+        std::vector<double> distances(data.size()*nn);
+
+        BOOST_FOREACH (const InterestPoint* ip, data){
+            ip->getDescriptor()->getFlatDescription(data_descriptors);
+        }
+
+        BOOST_FOREACH (const InterestPoint* ip, reference){
+            ip->getDescriptor()->getFlatDescription(reference_descriptors);
+        }
+
+        const size_t length = data_descriptors.size()/data.size();
+
+        flann::Matrix<double> flann_query(&data_descriptors[0], data.size(), length);
+        flann::Matrix<double> flann_reference(&reference_descriptors[0], reference.size(), length);
+        flann::Matrix<int> flann_indices(&indices[0], flann_query.rows, nn);
+        flann::Matrix<double> flann_distances(&distances[0], flann_query.rows, nn);
+
+        // construct a randomized kd-tree index using 4 kd-trees
+        flann::Index<flann::L2<double> > index(flann_reference, flann::KDTreeIndexParams(4));
```

Comment:
Maybe make the number of kd-trees a param/attribute; I'm just saying that if you put it on the constructor it'll be clear that there's a parameter for the algorithm, instead of having it here "hidden".